src/System/Random/Shuffle.hs

```diff
@@ -109,23 +112,27 @@ shuffle elements = shuffleTree (buildTree elements)
           in b : (shuffleTree rest rs)
 
 
--- |Given a sequence @(e1,...en)@ to shuffle, its length, and a random
--- generator, compute the corresponding permutation of the input
--- sequence.
-shuffle' :: RandomGen gen => [a] -> Int -> gen -> [a]
-shuffle' elements len = shuffle elements . rseq len
+-- |Given a sequence @(e1,...en)@ to shuffle and a random generator,
+-- compute the corresponding permutation of the input sequence.
+shuffleGen :: RandomGen gen => [a] -> gen -> ([a], gen)
```

Comment:
Exactly my thinking. Still unsure why the original `shuffle'` signature contained the length of the list